### PR TITLE
refactor: improve messenger list file structure and style

### PR DESCRIPTION
**Pull request by domw30**
### What does this do?
- use bemClassName and remove className attributes in messenger list files.
- one minor fix spotted whilst refactoring - selected users position/alignment.
- split each component into its own directory.
- extracted all styles from a massive single scss file into isolated component scss files in their own directory folders. 
- tidied up all imports and reduced duplication.
- manually tested local changes against development to prevent any breaks slipping through.

### Why are we making this change?
- remain consistent with style guide.
- this makes it 100x easier to navigate through the messenger list component styles.

### How do I test this?
Comparison with development:
Screenshot 2023-08-09 at 12 03 00

Screenshot 2023-08-09 at 12 02 11

Screenshot 2023-08-09 at 12 02 54

Screenshot 2023-08-09 at 12 02 36

#### Things to consider:
  1. How will this affect security?
  1. How will this affect performance?
  1. Does this change any APIs?

### CODE REVIEW

**Review by ratik21 — APPROVED** [review]

**Comment by domw30** [comment]
Thanks for the review @ratik21. I'll see what @dalefukami thinks as well before hitting the green button :)

File: src/components/messenger/list/conversation-item/conversation-item.tsx [hunk 1]
```diff
@@ -1,19 +1,23 @@
 import * as React from 'react';
```
Comment (dalefukami) [inline src/components/messenger/list/conversation-item/conversation-item.tsx:1]: We should probably stick to conventions and name this file `index.tsx`. Same for all the other ones.

**Review by dalefukami — APPROVED** [review]

File: src/components/messenger/list/conversation-item/conversation-item.tsx (hunk 1 above)
Comment (domw30) [reply src/components/messenger/list/conversation-item/conversation-item.tsx:1]: Okay. What would you name the `index.ts` file if renaming `conversation-item.tsx` to `index.tsx` ?

Comment (dalefukami) [reply src/components/messenger/list/conversation-item/conversation-item.tsx:1]: I wouldn't have one. None of our other components have one.

Comment (domw30) [reply src/components/messenger/list/conversation-item/conversation-item.tsx:1]: Okay no probs, the idea was an attempt to improve imports like 'import { menu } from './menu/menu`. I'll change those back. Thanks!

Comment (dalefukami) [reply src/components/messenger/list/conversation-item/conversation-item.tsx]: Yeah, the issue is that doesn't follow any of the preexisting conventions and it's a pretty significant change to the file structure. If we want to propose new conventions those should come in separate PRs where the merits of the new proposal can be discussed outside the distraction of other changes.

Comment (domw30) [reply src/components/messenger/list/conversation-item/conversation-item.tsx]: Nice yeah agree, doesn't make sense to have one component structured differently to the rest, understood that stirs up the existing conventions. Made those changes and checked everything is running okay :)

Comment (dalefukami) [reply src/components/messenger/list/conversation-item/conversation-item.tsx]: :+1: 

Love the PR by the way. Thanks for cleaning up!

**Pull request merged**